src/asl.ts

```diff
@@ -1446,6 +1525,100 @@ export class ASL {
     ) as any;
   }
 
+  /**
+   * Recursively evaluate a single expression, building a single {@link ASLGraph.NodeResults} object.
+   * All SubStates generated during evaluation will be merged into into a {@link ASLGraph.OutputSubState} along with the output
+   * of the handler callback.
+   *
+   * @param expr - Expression to evaluate.
+   * @param contextNode - Optional node to associate with the output state. This node may be used to name the resulting state.
```

Comment:
Does this doc comment properly apply to the overloads?

Reply:
Not the `contextNode` one... I guess I can duplicate the comments.